src/Features/CSharp/Portable/CodeRefactorings/ConvertLocalFunctionToMethod/CSharpConvertLocalFunctionToMethodCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,279 @@
+                typeParameters: typeParameters.ToImmutableArray(),
+                parameters: parameters.AddRange(capturesAsParameters));
+
+            var defaultOptions = CodeGenerationOptions.Default;
+            var method = MethodGenerator.GenerateMethodDeclaration(methodSymbol, CodeGenerationDestination.Unspecified,
+                    document.Project.Solution.Workspace, defaultOptions, root.SyntaxTree.Options);
+
+            var generator = CSharpSyntaxGenerator.Instance;
+            var editor = new SyntaxEditor(root, generator);
+
+            var needsRename = methodName != declaredSymbol.Name;
+            var identifierToken = needsRename ? methodName.ToIdentifierToken() : default;
+            var supportsNonTrailing = SupportsNonTrailingNamedArguments(root.SyntaxTree.Options);
+            var hasAdditionalArguments = !capturesAsParameters.IsEmpty();
+            var additionalTypeParameters = typeParameters.Except(declaredSymbol.TypeParameters).ToList();
+            var hasAdditionalTypeArguments = !additionalTypeParameters.IsEmpty();
+            var additionalTypeArguments = hasAdditionalTypeArguments
+                ? additionalTypeParameters.Select(p => (TypeSyntax)p.Name.ToIdentifierName()).ToArray()
+                : null;
+
+            // Update callers' name, arguments and type arguments
+            foreach (var node in parentBlock.DescendantNodes())
+            {
+                // A local function reference can only be an identifier or a generic name.
+                switch (node.Kind())
+                {
+                    case SyntaxKind.IdentifierName:
+                    case SyntaxKind.GenericName:
+                        break;
+                    default:
+                        continue;
+                }
+
+                // Using symbol to get type arguments, since it could be inferred and not present in the source
+                var symbol = semanticModel.GetSymbolInfo(node, cancellationToken).Symbol as IMethodSymbol;
+                if (symbol?.OriginalDefinition != declaredSymbol)
+                {
+                    continue;
+                }
+
+                var currentNode = node;
+
+                if (needsRename)
+                {
+                    currentNode = ((SimpleNameSyntax)currentNode).WithIdentifier(identifierToken);
+                }
+
+                if (hasAdditionalTypeArguments)
+                {
+                    var existingTypeArguments = symbol.TypeArguments.Select(x => x.GenerateTypeSyntax());
+                    // Prepend additional type arguments to preserve lexical order in which they are defined
+                    var typeArguments = additionalTypeArguments.Concat(existingTypeArguments);
+                    currentNode = generator.WithTypeArguments(currentNode, typeArguments);
+                }
+
+                if (node.Parent.IsKind(SyntaxKind.InvocationExpression, out InvocationExpressionSyntax invocation))
```

Comment:
still can't wait for your feature that looks for ```x.IsKind(Y); (YSyntax)x``` and converts it to this :)

Reply:
out of curiosity, where that should go to only be available in roslyn?